### PR TITLE
Porter: support bgp crd config

### PR DESCRIPTION
**Pull request by duanjiong**
/kind feature

Support bgp crd config, so we can dynamic add/update/delete bgp global/neighbor .

### CODE REVIEW

File: api/v1alpha1/eip_types.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type EipSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Pattern=`^([0-9]{1,3}\.){3}[0-9]{1,3}((\/([0-9]|[1-2][0-9]|3[0-2]))|(\-([0-9]{1,3}\.){3}[0-9]{1,3}))?$`
 	Address       string `json:"address,omitempty"`
+	Protocol      string `json:"protocol,omitempty"`
```
Comment (zheng1) [inline api/v1alpha1/eip_types.go:28]: Should we update the version to v1alpha2?

**Comment by zheng1** [comment]
Great works! If Protocol does not set default values, then we need to update the version of the EIP CRD.
And fix the CI error. https://github.com/kubesphere/porter/pull/98/checks?check_run_id=709215766#step:5:304

**Comment by duanjiong** [comment]
> Great works! If Protocol does not set default values, then we need to update the version of the EIP CRD.
> And fix the CI error. https://github.com/kubesphere/porter/pull/98/checks?check_run_id=709215766#step:5:304

the CI error is caused by iptables,  i will fix it later.

File: api/v1alpha1/eip_types.go (hunk 1 above)
Comment (duanjiong) [reply api/v1alpha1/eip_types.go:28]: the version update could be done later,  I have no idea on how it works now.

**Comment by duanjiong** [comment]
> Great works! If Protocol does not set default values, then we need to update the version of the EIP CRD.
> And fix the CI error. https://github.com/kubesphere/porter/pull/98/checks?check_run_id=709215766#step:5:304

If protocol is not specified,  it is same with bgp, and no special treatment is required.
What you mentioned above should be the API conversion.

**Comment by zheng1** [comment]
> > Great works! If Protocol does not set default values, then we need to update the version of the EIP CRD.
> > And fix the CI error. https://github.com/kubesphere/porter/pull/98/checks?check_run_id=709215766#step:5:304
> 
> If protocol is not specified, it is same with bgp, and no special treatment is required.
> What you mentioned above should be the API conversion.

Okay, I think the others are all right.

File: doc/layer2.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Porter Layer2 Usage
+Porter now support bgp and layer2, and will support non-bgp switch in future. In
+order to distinguish them, file `protocol` was added. 
+
+An example of how to use it will follow.
```
Comment (FeynmanZhou) [inline doc/layer2.md:5]: @duanjiong  Is there any prerequisites when using Layer2 with Porter?

Comment (duanjiong) [reply doc/layer2.md:5]: There is a config about kube-proxy,  i will doc about it later.

**Pull request merged**